packages/std/src/math/uint256.rs

```diff
@@ -206,8 +206,40 @@ impl Uint256 {
         ]
     }
 
-    pub fn is_zero(&self) -> bool {
-        self.0.is_zero()
+    pub const fn is_zero(&self) -> bool {
+        let bytes = self.to_be_bytes();
+        bytes[0] == 0
+            && bytes[1] == 0
+            && bytes[2] == 0
+            && bytes[3] == 0
+            && bytes[4] == 0
+            && bytes[5] == 0
+            && bytes[6] == 0
+            && bytes[7] == 0
+            && bytes[8] == 0
+            && bytes[9] == 0
+            && bytes[10] == 0
+            && bytes[11] == 0
+            && bytes[12] == 0
+            && bytes[13] == 0
+            && bytes[14] == 0
+            && bytes[15] == 0
+            && bytes[16] == 0
+            && bytes[17] == 0
+            && bytes[18] == 0
+            && bytes[19] == 0
+            && bytes[20] == 0
+            && bytes[21] == 0
+            && bytes[22] == 0
+            && bytes[23] == 0
+            && bytes[24] == 0
+            && bytes[25] == 0
+            && bytes[26] == 0
+            && bytes[27] == 0
+            && bytes[28] == 0
+            && bytes[29] == 0
+            && bytes[30] == 0
+            && bytes[31] == 0
```

Comment:
Here is a different implementation that is a bit shorter and more direct:

```rust
    pub const fn is_zero(&self) -> bool {
        let words = (self.0).0;
        words[0] == 0 && words[1] == 0 && words[2] == 0 && words[3] == 0
    }
```

The uint types are build up on u64 words which you can access. This is also how `to_be_bytes` works.

Reply:
Good point, done.